### DISCUSSION TITLE
Add a custom hyperlink in the gene description of Linear Genome View

### DISCUSSION THREAD

**Discussion by Akusem** [GMOD/jbrowse-components#1507] [General] [answered]
Hello,
I'm currently using JBrowseLinearView in my website to display basic genome information.
My problem is that I would like to add an hyperlink to my website's gene page when the user is clicking on the gene name (in the attributes field). 

For Example, would it be possible to defined a custom function that would filter/transform the field gene_name (in the picture below) ?

![image](https://user-images.githubusercontent.com/24433720/100640669-fa912980-3336-11eb-8c70-30a0b80fa95f.png)

I haven't seen code to do that in BaseFeatureDetail.tsx, but maybe it is possible ?

**Comment by cmdcolin** [comment, marked as answer]
Good question.

In jbrowse 1 we had this system "fmtFeatureDetail_fieldname" https://jbrowse.org/docs/mouse_configs.html#customizing-parts-of-the-view-details-pop-ups-with-callbacks but I always thought that was a bit hacky. We could try to add somethign sort of like that though but it does not yet exist in jbrowse 2

Note that we do render things that "look like HTML", which is run through the sanitize html package, so we can render some simple things like `<a>, <b>, <i> etc` but that implies sort of encoding literal HTML into your gff or similar which isn't that great

**Comment by Akusem** [comment]
Thanks for the answer,
I've tried to encode the HTML in the GFF3 files, but unfortunately it's impossible to set the tags attribute (and so the hyperlink) as the `=` sign is, I think, cut by Gff3TabixAdapter.

Example:
![image](https://user-images.githubusercontent.com/24433720/100742191-ad669380-33da-11eb-8067-2a432c1057b7.png)

I took another look at BaseFeatureDetail.tsx and contrary to what I thought, it was possible to pass a formatting function for the attributes. I guess the problem is to make the link with the config file and the BaseFeatureDetails rendered ?

**Reply by cmdcolin** [reply]
I think if you replace the = with %3D it can work https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md#the-gap-attribute

that could be a shortcut for now, but maybe in the future we will add a better way to do this

**Reply by Akusem** [reply]
Yes it work !
